GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java

```diff
@@ -100,7 +100,8 @@ protected void onCreate(Bundle savedInstanceState) {
                     mDelegate = (GVRActivityDelegate) aClass.newInstance();
                     mAppSettings = mDelegate.makeVrAppSettings();
                     mDelegate.onCreate(this);
-
+                    mActivityNative = mDelegate.getActivityNative();
```

Comment:
There is already code to invoke handleOnDock immediately if the backend requests it. GVRActivity.java:
```
    public void setMain(GVRMain gvrMain, String dataFileName) {
        this.mGVRMain = gvrMain;
        if (getRequestedOrientation() == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
            onConfigure(dataFileName);
            mDelegate.setMain(gvrMain, dataFileName);

            boolean isMonoscopicMode = mAppSettings.getMonoscopicModeParams().isMonoscopicMode();
            if (!isMonoscopicMode) {
                mViewManager = mDelegate.makeViewManager();
            } else {
                mViewManager = mDelegate.makeMonoscopicViewManager();
            }
            mDelegate.setViewManager(mViewManager);

            if (mConfigurationManager.isDockListenerRequired()) {
                startDockEventReceiver();
            } else {
                handleOnDock();
            }
```

So this is not needed.